src/structures/MessageActionRow.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const BaseMessageComponent = require('./BaseMessageComponent');
+const { MessageComponentTypes } = require('../util/Constants');
+
+/**
+ * Represents an ActionRow containing message components.
+ */
+class MessageActionRow extends BaseMessageComponent {
+  /**
+   * @typedef {BaseMessageComponentOptions} MessageActionRowOptions
+   * @property {MessageComponent[]|MessageComponentOptions[]} [components] The components to place in this ActionRow
+   */
+
+  /**
+   * @param {MessageActionRow|MessageActionRowOptions} [data={}] MessageActionRow to clone or raw data
+   */
+  constructor(data = {}) {
+    super({ type: 'ACTION_ROW' });
+
+    this.components = (data.components ?? []).map(c => BaseMessageComponent.create(c, null, true));
+  }
+
+  /**
+   * Adds a component to the row (max 5).
+   * @param {MessageComponent|MessageComponentOptions} component The component to add
+   * @returns {MessageActionRow}
+   */
+  addComponent(component) {
+    return this.addComponents({ ...component });
```

Comment:
Is the object spreading here intentional?

Reply:
Nope, not sure how that got introduced. Will remove